internal/locate/region_request.go

```diff
@@ -922,7 +983,17 @@ func (s *RegionRequestSender) SendReqCtx(
 			return nil, nil, errors.Trace(err)
 		}
 		if regionErr != nil {
-			retry, err = s.onRegionError(bo, rpcCtx, req, regionErr, &opts)
+			// Stale Read request will retry the leader or next peer on error,
+			// if txnScope is global, we will only retry the leader by using the WithLeaderOnly option,
+			// if txnScope is local, we will retry both other peers and the leader by the strategy of replicaSelector.
+			if tryTimes < 1 && req != nil && req.TxnScope == oracle.GlobalTxnScope && req.GetStaleRead() {
+				accessFollowerState, ok := s.replicaSelector.state.(*accessFollower)
+				if ok {
+					WithLeaderOnly()(&accessFollowerState.option)
+				}
+			}
```

Comment:
As stale read has some special logics, and `storeSelectorOp` is only for stale read now, actually I am considering if it's better to give it a special state in the `replicaSelector`...

Reply:
How about adding a field `isGlobalStaleRead` to the `accessFollower` state to distinguish it since the strategy of local stale read is same as follower read? If the `isGlobalStaleRead` is true, the second `next` call will try the leader.